src/Mod/PartDesign/App/FeatureLoft.cpp

```diff
@@ -72,7 +72,9 @@ Loft::getSectionShape(const char *name,
                       size_t expected_size)
 {
     std::vector<TopoShape> shapes;
-    if (subs.empty() || std::find(subs.begin(), subs.end(), std::string()) != subs.end()) {
+    auto subName = subs.empty() ? "" : subs.front();
+    // Be smart. If part of a sketch is selected, use the entire sketch unless it is a single vertex - backward compatibility (#16630)
+    if (subs.empty() || std::find(subs.begin(), subs.end(), std::string()) != subs.end() || ( obj->isDerivedFrom(Part::Part2DObject::getClassTypeId()) &&  subName.compare(0, 6, "Vertex") != 0)) {

```

Comment:
As you probably now see with the linter run, this triggers a magic number warning.

Also, this line is fairly cognitively complex.  I suggest a slight rewrite like
```
    // Be smart. If part of a sketch is selected, use the entire sketch unless it is a single vertex - 
    // backward compatibility (#16630)
    auto useEntireSketch = obj->isDerivedFrom(Part::Part2DObject::getClassTypeId()) &&  subName.find("Vertex") != 0;
    if (subs.empty() || std::find(subs.begin(), subs.end(), std::string()) != subs.end() || useEntireSketch ) {

```

That's opinion based on past feedback from maintainers; might be fine as it is.

Reply:
No classids for those, unfortunately (IFAIK),  There's a fair amount of explicit looking for "Face", "Edge" and "Vertex" in various places, so this is normal.  You may well already know this, but here's a useful jumping off point: https://old.opencascade.com/doc/occt-7.5.0/overview/html/occt_user_guides__modeling_data.html#occt_modat_5   And there are  FreeCAD Forums, Discord, and Telegram all out there if you want another way to contact people.  I do not frequent the forums, but I'm around on the other two a lot.